src/control/lib/telemetry/counter.go

```diff
@@ -0,0 +1,73 @@
+//
+// (C) Copyright 2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+// +build linux,amd64
+//
+
+package telemetry
+
+/*
+#cgo LDFLAGS: -lgurt
+
+#include "gurt/telemetry_common.h"
+#include "gurt/telemetry_consumer.h"
+*/
+import "C"
+
+import (
+	"context"
+)
+
+type Counter struct {
+	metricBase
+}
+
+func (c *Counter) Type() MetricType {
+	return MetricTypeCounter
+}
+
+func (c *Counter) FloatValue() float64 {
+	return float64(c.Value())
+}
+
+func (c *Counter) Value() uint64 {
```

Comment:
Rather than cluttering up the API with the need for error checking (because really, how do you handle an error when you couldn't fetch a metric value?), I'd like to try returning max values instead. I agree that it's not very helpful to hide an error with 0 as that could be a legitimate value, but returning MAX_UINT will get attention when the graphs go wacky (as they did when I was collecting stats that hadn't been computed yet). One could argue that the caller would want to drop bad values rather than collecting them, and this is a case for returning an error, but in practice I think that would also lead to weirdness as you'd get "holes" in the data.